docs/design/networking.md

```diff
@@ -97,11 +99,18 @@ So we end up with 3 cases:
 
 1. Container -> Container or Container <-> VM. These should use 10. addresses directly and there should be no NAT.
 
-2. Container -> Internet. These have to get mapped to the primary host IP so that GCE knows how to egress that traffic. There is actually 2 layers of NAT here: Container IP -> Internal Host IP -> External Host IP. The first level happens in the guest with IP tables and the second happens as part of GCE networking. The first one (Container IP -> internal host IP) does dynamic port allocation while the second maps ports 1:1.
+2. Container -> Internet. These have to get mapped to the primary host IP so that the host node knows how to egress that traffic. In most networking setups there are actually 2 layers of NAT here: Container IP -> Internal Host IP -> External Host IP. The first level happens in the guest with IP tables and the second happens as part of egress past a firewall to the broader internet.  On cloud providers, this is usually provided by the cloud virtual network.  On physical providers, this is provided by some sort of gateway router.. The first one (Container IP -> internal host IP) does dynamic port allocation while the second maps ports 1:1.
 
-3. Internet -> Container. This also has to go through the primary host IP and also has 2 levels of NAT, ideally. However, the path currently is a proxy with (External Host IP -> Internal Host IP -> Docker) -> (Docker -> Container IP). Once [issue #15](https://github.com/GoogleCloudPlatform/kubernetes/issues/15) is closed, it should be External Host IP -> Internal Host IP -> Container IP. But to get that second arrow we have to set up the port forwarding iptables rules per mapped port.
+3. Internet -> Container. This also has to go through the primary host IP and has 1 level of NAT, ideally. However, the path currently is a proxy with (External Host IP -> Internal Host IP -> Docker) -> (Docker -> Container IP). Once [issue #15](https://github.com/GoogleCloudPlatform/kubernetes/issues/15) is closed, it should be External Host IP -> Container IP. The way that this will work is that each
+container that wants to be present on the external internet will specify an external ip address.  The IP address will be trapped via IPTables on the host node and re-written to the container IP address.
```

Comment:
the iptables is purely for external ip -> internal ip rewriting.  Even if you have ipvlan, I don't think that you can map the external IP to the pod without iptables, unless you somehow make the bridge that the container allocates addresses out of contain external IP addresses, but that seems pretty tricky (and dangerous) you're always going to want something that NATs the packet from external to internal IP.  I suppose you could do that at the edge instead, if you want to...